modules/core/src/core/animation-loop.js

```diff
@@ -227,7 +240,9 @@ export default class AnimationLoop {
 
     this.redraw();
 
-    // Request another render frame now
+    // Request another render frame now but cancel any pending renders
+    // to ensure only one loop can ever run
+    cancelAnimationFrame(this._animationFrameId);
```

Comment:
Is this necessary here?

Reply:
It's not necessary as long as `_renderFrame` is only ever called by itself or `_startLoop` (which also contains the check). This is currently the case, but I'm not sure we can guarantee this going forward. I'm generally against defense coding measures like this, but in this case I've found it to be useful for two reasons:

1. It protects against a really easy-to-introduce / hard-to-discover bug. (I only discovered that we had multiple loops running in our application because I happened to be reading and reasoning about the AnimationLoop logic here.)
2. It communicates to readers the (perhaps self-evident) intent that this is meant to be a "singleton" loop.

One option is to move this `_renderFrame` function into `startLoop`, a pattern I've seen quite a bit. That would allow us to drop that extra check and also make sure `_renderFrame` is only called in this one spot. Something like:

```js
_startLoop () {
  const renderFrame = () => {
    if (!this._running) return
    this.redraw()
    this._animationFrameId = requestAnimationFrame(renderFrame)
  }
  cancelAnimationFrame(this._animationFrameId)
  this.animationFrameId = requestAnimationFrame(renderFrame)
}
```

Thoughts?

Reply:
From offline convo with Tarek, let's apply this change as part of this PR.